src/Aspire.Hosting.Garnet/GarnetBuilderExtensions.cs

```diff
@@ -0,0 +1,131 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Garnet;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Garnet resources to the application model.
+/// </summary>
+public static class GarnetBuilderExtensions
+{
+    /// <summary>
+    /// Adds a Garnet container to the application model.
+    /// </summary>
+    /// <example>
+    /// <remarks>Use in AspireHost</remarks>
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api") 
+    /// var garnet = builder.AddGarnet("garnet"); 
+    /// api.WithReference(garnet);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// <remarks>Use in Api with Aspire.StackExchange.Redis</remarks>
+    /// <code>
+    /// var builder = WebApplication.CreateBuilder(args);
+    /// builder.AddRedisClient("garnet");
+    ///
+    /// var multiplexer = builder.Services.BuildServiceProvider()
+    ///                                   .GetRequiredService&lt;IConnectionMultiplexer&gt;();
+    /// 
+    /// var db = multiplexer.GetDatabase();
+    /// db.HashSet("key", [new HashEntry("hash", "value")]);
+    /// var value = db.HashGet("key", "hash");
+    /// </code>
+    /// </example>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port to bind the underlying container to.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<GarnetResource> AddGarnet(this IDistributedApplicationBuilder builder, string name, int? port = null)
```

Comment:
How do we think of Garnet with regard to experimental? It is itself a research project, should we annotate this extension method with experimental?

Reply:
I'm not sure it should be experimental.

Reply:
Really. It was experimental when they created the radish resource